postgrenosql/src/main/java/com/yahoo/ycsb/db/PostgreNoSQLDBClient.java

```diff
@@ -0,0 +1,303 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License. See accompanying
+ * LICENSE file.
+ */
+package com.yahoo.ycsb.db;
+
+import com.yahoo.ycsb.*;
+import org.json.simple.JSONObject;
+import org.postgresql.Driver;
+import org.postgresql.util.PGobject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.*;
+
+/**
+ * PostgreNoSQL client for YCSB framework.
+ */
+public class PostgreNoSQLDBClient extends DB {
+  private static final Logger LOG = LoggerFactory.getLogger(PostgreNoSQLDBClient.class);
+
+  /** The driver to get the connection to postgresql. */
+  private static Driver postgrenosqlDriver;
+
+  /** The connection to the database. */
+  private static Connection connection;
+
+  /** The class to use as the jdbc driver. */
+  public static final String DRIVER_CLASS = "db.driver";
+
+  /** The URL to connect to the database. */
+  public static final String CONNECTION_URL = "postgrenosql.url";
+
+  /** The user name to use to connect to the database. */
+  public static final String CONNECTION_USER = "postgrenosql.user";
+
+  /** The password to use for establishing the connection. */
+  public static final String CONNECTION_PASSWD = "postgrenosql.passwd";
+
+  /** The JDBC connection auto-commit property for the driver. */
+  public static final String JDBC_AUTO_COMMIT = "postgrenosql.autocommit";
+
+  /** The primary key in the user table. */
+  public static final String PRIMARY_KEY = "YCSB_KEY";
+
+  /** The field name prefix in the table. */
+  public static final String COLUMN_NAME = "YCSB_VALUE";
+
+  private static final String DEFAULT_PROP = "";
+
+  private boolean initialized = false;
+  private Properties props;
+
+  /** Returns parsed boolean value from the properties if set, otherwise returns defaultVal. */
+  private static boolean getBoolProperty(Properties props, String key, boolean defaultVal) {
+    String valueStr = props.getProperty(key);
+    if (valueStr != null) {
+      return Boolean.parseBoolean(valueStr);
+    }
+    return defaultVal;
+  }
+
+  @Override
+  public void init() throws DBException {
+    if (initialized) {
+      LOG.error("Client connection already initialized.");
+      return;
+    }
+    props = getProperties();
+    String urls = props.getProperty(CONNECTION_URL, DEFAULT_PROP);
+    String user = props.getProperty(CONNECTION_USER, DEFAULT_PROP);
+    String passwd = props.getProperty(CONNECTION_PASSWD, DEFAULT_PROP);
+    boolean autoCommit = getBoolProperty(props, JDBC_AUTO_COMMIT, true);
+
+    try{
+      Properties tmpProps = new Properties();
+      tmpProps.setProperty("user", user);
+      tmpProps.setProperty("password", passwd);
+
+      postgrenosqlDriver = new Driver();
```

Comment:
this method is going to get called on an instance per client thread, so these static variables will need locking to guard them.

A few of the other bindings have examples of doing this, as well as using an AtomicInteger to reference count.